src/application/sidebar/FileExplorer.tsx

```diff
@@ -51,29 +73,9 @@ export function FolderNode({ folder }: FolderNodeProps) {
   const [collapsed, setCollapsed] = useAtom(folder.collapsedAtom);
 
   return (
-    <div className="flex flex-col items-start gap-2 self-stretch">
-      <div
-        className={cx('flex cursor-pointer select-none items-start gap-1 self-stretch')}
-        onClick={() => setCollapsed((currentState) => !currentState)}
-      >
-        <div
-          className={cx('text-btn-ico-side-bar-item transition duration-100 ease-in-out', { 'rotate-90': !collapsed })}
-        >
-          <ArrowRightIcon />
-        </div>
-        <h2 className="overflow-hidden overflow-ellipsis whitespace-nowrap">{folder.name}</h2>
-      </div>
-      {!collapsed && (
-        <div className="flex flex-col items-start gap-2 self-stretch pl-5">
-          {childrenEntries.map((child) =>
-            child.isFolder ? (
-              <FolderNode folder={child} key={child.path} />
-            ) : (
-              <FileNode existingFileNames={childrenEntries.map((file) => file.name)} file={child} key={child.path} />
-            ),
-          )}
-        </div>
-      )}
+    <div className="flex flex-col items-start gap-1 self-stretch">
+      <ProjectExplorerChevronNode collapsed={collapsed} text={folder.name} onClick={() => setCollapsed((v) => !v)} />
+      <ProjectExplorerEntries collapsed={collapsed} projectFileEntries={childrenEntries} />
```

Comment:
Could you do `{!collapsed && <ProjectExplorerEntries...}` instead? And remove `collapsed` from `ProjectExplorerEntries` props